src/components/Sidebar.js

```diff
@@ -32,7 +32,7 @@ class Sidebar extends Component {
 
   handleDownload() {
     const { gifData } = this.props;
-    download(gifData, 'gifsmos.gif', 'image/gif');
+    download(gifData, this.props.gifFileName || 'gifsmos.gif', 'image/gif');
```

Comment:
since you're destructuring props on L34, i'd just extract `gifFileName` as well, so that you can remove `this.props` on this line.